Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1563,6 +1563,15 @@ extension BrowserViewController: WKNavigationDelegate {
             // VoiceOver will sometimes be stuck on the element, not allowing user to move
             // forward/backward. Strange, but LayoutChanged fixes that.
             UIAccessibilityPostNotification(UIAccessibilityLayoutChangedNotification, nil)
+        } else {
+            // Tab is in the backgroud, but we want to be able to see it in the TabTray.
+            // Delay 100ms to let the tab render (it doesn't work without the delay)
+            // before screenshotting.
+            let time = dispatch_time(DISPATCH_TIME_NOW, Int64(100 * NSEC_PER_MSEC))
+            dispatch_after(time, dispatch_get_main_queue()) {
+                let screenshot = self.screenshotHelper.takeScreenshot(tab, aspectRatio: 0, quality: 1)
+                tab.setScreenshot(screenshot)
+            }
```

Comment:
You can also create a serial `NSOperationQueue` for this and let the screenshotting run on there in the background. Then they will just go one-by-one and as fast as possible without delays in between.

If you put the screenshot logic into a `ScreenshotTask` class, with a *weak* reference to the `tab` then you can prevent taking screenshots from tabs that may not be there anymore.

This could work really well combined with using a notification to signal the `TabTrayController` that a new screenshot ready.